src/sonic-build-hooks/scripts/buildinfo_base.sh

```diff
@@ -15,26 +15,27 @@ REPR_MIRROR_URL_PATTERN='http:\/\/packages.trafficmanager.net\/'
 DPKG_INSTALLTION_LOCK_FILE=/tmp/.dpkg_installation.lock
 
 . $BUILDINFO_PATH/config/buildinfo.config
+
+if [ "$(whoami)" != "root" ] && [ -n "$(which sudo)" ];then
+    SUDO=sudo
+else
+    SUDO=''
+fi
+
 if [ -e /vcache ]; then
 	PKG_CACHE_PATH=/vcache/${IMAGENAME}
 else
 	PKG_CACHE_PATH=/sonic/target/vcache/${IMAGENAME}
 fi
 PKG_CACHE_FILE_NAME=${PKG_CACHE_PATH}/cache.tgz
-sudo chown $USER $(dirname $PKG_CACHE_PATH)
+$SUDO chown $USER $(dirname $PKG_CACHE_PATH)
```

Comment:
@liushilongbuaa do we need to put it AFTER `mkdir`?

Reply:
fixed.